runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -307,6 +307,37 @@ impl Verifier {
             )));
         }
 
+        // Verify our own RAK is published in registry.
```

Comment:
Only do this once per epoch. Also remember the found node ID and only lookup that entry in subsequent checks.

Reply:
good ideas, added.